crates/autopilot/src/solvable_orders.rs

```diff
@@ -381,11 +403,14 @@ async fn filter_banned_user_orders(
             ]
         }))
         .await;
-    orders.retain(|order| {
-        !banned.contains(&order.metadata.owner)
-            && !banned.contains(&order.data.receiver.unwrap_or_default())
-    });
     orders
+        .iter()
+        .filter_map(|order| {
+            (banned.contains(&order.metadata.owner)
+                || banned.contains(&order.data.receiver.unwrap_or_default()))
+            .then_some(order.metadata.uid)
```

Comment:
nit: Same suggestion as above to avoid the map lookup when `receiver == None`.
```suggestion
            std::iter::once(&order.metadata.owner)
                .chain(&order.data.receiver)
                .any(|addr| banned.contains(addr))
                .then_some(order.metadata.uid)
```